packages/react-server-redux/ReduxAdapter.js

```diff
@@ -0,0 +1,60 @@
+const Q = require('q');
+
+module.exports = class ReduxAdapter {
+	constructor(store) {
+		this.stateWaitMap = {};
+		this.store = store;
+	}
+
+	//Search the state for the values we are waiting on
+	//State is considered ready when the waiting value is neither
+	//undefined or null
+	checkStateStatus() {
+		const state = this.store.getState();
+		const stateKeys = Object.keys(this.stateWaitMap);
+
+		for (let i = 0; i < stateKeys.length; i++) {
```

Comment:
Maybe it would be easier to read if you used a `forEach`?

```
Object.keys(this.stateWaitMap).forEach(stateKey => {
  if (stateKey.promise.isPending()) {
```